lib/src/multires_stochastic_texture_synthesis.rs

```diff
@@ -1389,8 +1388,9 @@ impl TreeGrid {
                         + (y as i64 - place_to_look.closest_point_on_boundary_y)
                             * (y as i64 - place_to_look.closest_point_on_boundary_y);
 
-                    if squared_distance_to_closest_possible_point_on_chunk
-                        > upper_bound_kth_best_squared_distance
+                    if !tmp_result.is_empty()
```

Comment:
I don't like the unwraps I added here and below but I'm not sure of a more elegant way to do this.